user/test/com/google/gwt/user/client/ui/ImageTest.java

```diff
@@ -790,10 +790,25 @@ public void testWrapOfSubclass() {
     assertNotNull(image);
 
     // Cleanup.
-    Document.get().getBody().appendChild(div);
```

Comment:
I actually thought that it should be `removeChild()` and that the test would then cover that `detachNow()` does not fail if the element is already removed from the DOM. Reason is that you usually call `detachNow` AFTER the element has been removed from the DOM through some other means, just like the JavaDoc of `wrap()` states.

Same for the test you added with calling `removeFromParent()`. It should not fail if the Widget element isn't in the DOM anymore.

Reply:
Ah, that makes more sense. I'll add a test so that we also cover the "detach even though it is connected" case, as the test used to do. That historically worked (since the old test was adding twice instead of never removing).

Reply:
All tests passing https://github.com/niloc132/gwt/actions/runs/11038034684